util/uuid.h

```diff
@@ -0,0 +1,150 @@
+// https://en.wikipedia.org/wiki/Universally_unique_identifier
+//
+// This is considered suitable for identifying medium-high frequency events.
+// For example, if 1 million computers each generate 10 random RfcUuids every
+// second for a year, the chance of one being repeated in that time is 1 in
+// millions. See Env::GenerateRfcUuid().
+//
+// Implementation detail: RocksDB uses a byte ordering that puts the
+// 4-bit version in the upper-most bits of upper64 and the 1-4 bit variant
+// in the upper-most bits of lower64.
+struct RfcUuid {
+  RawUuid data = 0;
+
+  // Parses from the 36-character form with hexadecimal digits and four
+  // hyphens. Returns Corruption if malformed.
+  static Status Parse(const Slice &input, RfcUuid *out);
+
+  // Assuming the input is random with 128 bits of entropy, this converts to
+  // an RfcUuid by setting variant 1 version 4, losing 6 bits of entropy.
+  static RfcUuid FromRawLoseData(const RawUuid &raw);
+
+  // Generates the 36-character text form
+  std::string ToString() const;
+  // Generates the 36-character text form directly into a buffer,
+  // returning a Slice into the buffer for the 36 characters written.
+  Slice PutString(char *buf_of_36) const;
+
+  // Should only be true when unset
+  inline bool IsEmpty() { return data == 0; }
+
+  // From RFC 4122, except reserved/unspecified variants are unchecked and
+  // might be returned. Variant is >= 0 and <= 4
+  int GetVariant() const;
+  // Sets variant while leaving other data bits in place. Note that higher
+  // variants overwrite more data bits, so the variant should generally only
+  // be set once.
+  void SetVariant(int variant);
+
+  // From RFC 4122, Version is >= 0 and <= 15
+  int GetVersion() const;
+  // Sets version while leaving other data bits in place.
+  void SetVersion(int version);
+};
+
+inline bool operator<(const RfcUuid &lhs, const RfcUuid &rhs) {
+  return lhs.data < rhs.data;
+}
+
+inline bool operator==(const RfcUuid &lhs, const RfcUuid &rhs) {
+  return lhs.data == rhs.data;
+}
+
+// A "mostly unique ID" encodable into 20 characters in base 36 ([0-9][A-Z])
+// This provides a compact, portable text representation with 103.4 bits of
+// entropy. This was originally developed for DB Session IDs (see
+// DB::GetDbSessionId(). All zeros means unset or empty.
+//
+// This is considered suitable for identifying medium-frequency events or
+// narrow scope events. For example, if 1 million computers each generate a
+// random RocksMuid every minute for a year, or one computer generates a
+// million every second for a week, the chance of a RocksMuid being repeated
+// in that time is 1 in millions. See Env::GenerateRawUuid() and
+// FromRawLoseData().
+//
+// In text form, the digits are in standard written order: lowest-order last,
+// except each piece (lower64 and upper64) only holds 10 base-36 digits.
+// Thus, each uint64_t piece is < kPieceMod. FIXME
+struct RocksMuid {
+  RawUuid scaled_data = 0;
```

Comment:
Do you mean the scaled_data to be public?  Should users have free access to it and be able to change it?